src/main/java/com/google/cloud/tools/app/deploy/gcloud/GcloudFuture.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.app.deploy.gcloud;
+
+import com.google.cloud.tools.app.deploy.process.OutputHandler;
+import com.google.cloud.tools.app.deploy.process.StreamConsumer;
+import com.google.common.base.Joiner;
+import com.google.common.collect.Lists;
+
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class GcloudFuture implements Future<String> {
+
+  // does this need to be synchronized some how when storing the result?
+
+  private List<String> result = Lists.newArrayList();
+  private final Process process;
+  private boolean cancelled = false;
+
+  public GcloudFuture(Process process) {
```

Comment:
I guess we could potentially drop off the management of the Process to the ExecutorService instead of managing it ourselves with shutdown hooks, etc.